src/unix/signal.c

```diff
@@ -63,6 +68,20 @@ static void uv__signal_global_init(void) {
 }
 
 
+static void uv__signal_global_reinit(void) {
+  /* We can only use signal-safe functions here.
+   * That includes read/write and close, fortunately.
+   * We do all of this directly here instead of resetting
+   * uv__signal_global_init_guard because
+   * it'uv__signal_global_once_init is only called from uv_loop_init
+   * and this needs to function in existing loops.
+   */
+  uv__close(uv__signal_lock_pipefd[0]);
+  uv__close(uv__signal_lock_pipefd[1]);
+  uv__signal_global_init();
```

Comment:
please set these to -1.  i know we'll set them to something else next, but if something fails I want to see -1 and not some old value.